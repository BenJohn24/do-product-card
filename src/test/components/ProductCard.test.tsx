import { ProductCard } from "../../components";
import { render, screen, fireEvent } from '@testing-library/react';
import { product1 } from "../data/products";



describe('ProductCard', () => {

    test('debe de mostrar el component con el nombre del producto', () => {

      const { asFragment } = render(<ProductCard  product={ product1}>
        {
          () => (
            <h1>Product Card</h1>
          )
        }
        </ProductCard>
      
      )
      expect(asFragment()).toMatchSnapshot();
    });

    test('debe de incrementar el contador', () => {
      
      const { asFragment } = render(<ProductCard  product={ product1}>
        {
          ({ count, increaseBy }) => (
            <>
            
            <h1>Product Card</h1>
            <span>Count: { count }</span>
            <button onClick={ () => increaseBy(1)}>+1</button>
            </>
          )
        }
        </ProductCard>
      
      )
      
    });

    render(<ProductCard  product={ product1}>
      {
        ({ count, increaseBy }) => (
          <>
          
          <h1>Product Card</h1>
          <span> { count }</span>
          <button onClick={ () => increaseBy(1)}>+1</button>
          </>
        )
      }
      </ProductCard>)

    const button = screen.getByRole('button');
    const counterText = screen.getByText('0');
    fireEvent.click(button);

    expect(screen.getByText('1')).toBeInTheDocument();




    
    
})