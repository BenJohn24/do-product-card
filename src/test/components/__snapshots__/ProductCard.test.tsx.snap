// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard > debe de mostrar el component con el nombre del producto 1`] = `
<DocumentFragment>
  <div
    class="_productCard_133f98 undefined"
  >
    <h1>
      Product Card
    </h1>
  </div>
</DocumentFragment>
`;
